ibtk/include/ibtk/IBTK_MPI.h

```diff
@@ -0,0 +1,332 @@
+//
+//    * Redistributions of source code must retain the above copyright notice,
+//      this list of conditions and the following disclaimer.
+//
+//    * Redistributions in binary form must reproduce the above copyright
+//      notice, this list of conditions and the following disclaimer in the
+//      documentation and/or other materials provided with the distribution.
+//
+//    * Neither the name of The University of North Carolina nor the names of
+//      its contributors may be used to endorse or promote products derived from
+//      this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#ifndef included_IBTK_MPI
+#define included_IBTK_MPI
+
+/////////////////////////////// INCLUDES /////////////////////////////////////
+
+#include "SAMRAI_config.h"
+
+#include <mpi.h>
+
+#include "tbox/Utilities.h"
+
+namespace IBTK
+{
+inline MPI_Datatype
+mpi_type_id(const double)
+{
+    return MPI_DOUBLE;
+}
+
+inline MPI_Datatype
+mpi_type_id(const std::pair<double, int>)
+{
+    return MPI_DOUBLE_INT;
+}
+
+inline MPI_Datatype
+mpi_type_id(const int)
+{
+    return MPI_INT;
+}
+
+inline MPI_Datatype
+mpi_type_id(const std::pair<int, int>)
+{
+    return MPI_2INT;
+}
+
+inline MPI_Datatype
+mpi_type_id(const float)
+{
+    return MPI_FLOAT;
+}
+
+inline MPI_Datatype
+mpi_type_id(const std::pair<float, int>)
+{
+    return MPI_FLOAT_INT;
+}
+
+inline MPI_Datatype
+mpi_type_id(const char)
+{
+    return MPI_CHAR;
+}
+
+template <typename T>
+inline MPI_Datatype
+mpi_type_id(const T&)
+{
+    static_assert(!std::is_same<T, T>::value,
+                  "The given type does not have a corresponding MPI_Datatype value. At this time only char, int, "
+                  "unsigned int, float, double, std::pair<int, int>, std::pair<int, double>, and "
+                  "std::pair<int, float> are supported by IBTK_MPI.");
+    return MPI_CHAR;
+}
+
+/**
+ * @brief Provides C++ wrapper around MPI routines.
+ *
+ * The IBTK_MPI struct provides simple interfaces to common MPI routines. All function calls allow a user-provided
+ * communicator to be used. If a null communicator or none is present, the default communicator set by setCommunicator()
```

Comment:
sorry, one more thing: this is out of sync with the previous changeset. I would just delete the sentence "If a null communicator..." since the default argument is already present in the declaration.